docs/source/docs/index.rst

```diff
@@ -161,6 +161,7 @@ Read the research:
    ../serving/sky-serve
    ../serving/user-guides
    ../serving/service-yaml-spec
+   ../serving/https
```

Comment:
Should we have it at the top level of the doc or in the `user-guides`?

Reply:
This feels more like a serving dedicated guide to me as putting it in other places is possible to left the impression that this doc is for https on a cluster with ports exposed.. wdyt?